### DISCUSSION TITLE
blog/devenv-compose-developer-environment-for-php-with-nix/

### DISCUSSION THREAD

**Discussion by giscus** [shyim/shyim.me#48] [Q&A]
# blog/devenv-compose-developer-environment-for-php-with-nix/

I have been using Nix for almost four years and also maintaining some packages at nixpkgs (the primary Nix repository). The Nix learning curve is complicated, so I could not get my colleagues into it. But Devenv made it so easy to compose a developer environment, so I got their attraction, and some of them switched already, and we are looking to where we can use it too.
What is Devenv?

https://shyim.me/blog/devenv-compose-developer-environment-for-php-with-nix/

**Comment by raiden076** [comment]
Great! Thanks...

**Comment by souhaiebtar** [comment]
thank you, but
```
{ pkgs, ... }: {
   ...
   
  languages.php.package = pkgs.php81.buildEnv {
+   extensions = { all, enabled }: with all; enabled ++ [ redis ];
    extraConfig = ''
      memory_limit = 256m
    '';
  };
}
``` 
won't work, i tested it on wsl2 (new install ) and ubuntu 22.04, memory limit and redis not taked into consideration,

thank you anyway

**Reply by shyim** [reply]
In the meanwhile it got a lot of easier:

```nix
{ pkgs, ... }: {
   ...

  languages.php.version = "8.1";
  languages.php.extensions = [ "redis" ];
  languages.php.ini = ''
   memory_limit = 256m
   '';
}
```

**Reply by souhaiebtar** [reply]
thank you for the quick response, i did open the `devenv` website in hope of finding something in the documentation but nothing (or maybe it was there but i didn't see it), thanks again and have a great week-end

**Comment by souhaiebtar** [comment]
thank you, but still not working

**Reply by shyim** [reply]
can you post your complete configuration files?

**Reply by souhaiebtar** [reply]
```
{ pkgs, ... }:

{
  # https://devenv.sh/basics/
  env.GREET = "devenv";

  # https://devenv.sh/packages/
  packages = [ pkgs.git pkgs.php81 ];


  languages.php.version = "8.1";
  languages.php.extensions = [ "redis" ];
  languages.php.ini = ''
   memory_limit = 256m
  '';
}
```

**Reply by souhaiebtar** [reply]
thank you, i was able to make it work using the php example from the devenv repo,
the link is [php-caddy](https://github.com/cachix/devenv/blob/main/examples/caddy-php/devenv.nix)
